drivers/vulkan/rendering_device_vulkan.cpp

```diff
@@ -2874,6 +2874,32 @@ bool RenderingDeviceVulkan::texture_is_valid(RID p_texture) {
 	return texture_owner.owns(p_texture);
 }
 
+RD::TextureFormat RenderingDeviceVulkan::texture_get_format(RID p_texture) {
+	_THREAD_SAFE_METHOD_
+
+	Texture *tex = texture_owner.get_or_null(p_texture);
+	ERR_FAIL_COND_V(!tex, TextureFormat());
+
+	TextureFormat tf;
+
+	// Q? I guess we need to adjust this if this is a shared texture
```

Comment:
This may not be the right thing to do here but we may at least need to add a `texture_get_owner` method when we're dealing with a shared texture.